modules/custom/moj_resources/src/Plugin/rest/resource/ContentResource.php

```diff
@@ -156,6 +167,23 @@ protected function checkContentIdParameterIsNumeric()
             404
         );
     }
+
+    protected function checkPrisonIsNumeric()
+    {
+        if (is_numeric($this->paramater_prison)) {
+            return true;
+        }
+        throw new NotFoundHttpException(
+            t('The prison parameter must be a numeric'),
+            null,
+            404
```

Comment:
Should this be `400`/`BadRequestHttpException` to be consistent with the message on L177?

Reply:
I just pulled the function from another `*Resource.php` file 😆

Reply:
If you look on the front-end you'll see it 404s when you can't view a piece of content so don't really want to change this until we know why it's a 404....lovely code we have 😆

Reply:
Actually it'd be a very, very rare situation where this wouldn't be numeric since the `setPrison()` function pretty much ensures it is.

Reply:
This makes me a little uneasy, because the response we're introducing isn't "correct" even though it is consistent. If we caught `4xx`in the frontend and continued to show `404` to the user, would that allow us to retain the user-facing behaviour _and_ enable us to return more meaningful status codes from the Drupal API? It would also help us progressively improve the Drupal API to get us out of the technical debt we're in and make it easier/nicer to work with 🥳 .

I can see it being useful to capture these errors (with something like Sentry), because we shouldn't ever see a non-numeric prison, but a `404` for a piece of content is entirely predictable and wouldn't be an error state.

Reply:
Aye, changing things like this needs proper time and understanding the code to make sure changing the response code doesn't break the frontend.

Totally agree on it being done properly....including renaming all the paramaters to parameters 😆